geopandas/tests/test_clip.py

```diff
@@ -199,28 +199,28 @@ def test_clip_multipoly_keep_slivers(multi_poly_gdf, single_rectangle_gdf):
         assert "GeometryCollection" in clip.geom_type[0]
 
 
-def test_clip_multipoly_drop_slivers(multi_poly_gdf, single_rectangle_gdf):
+def test_clip_multipoly_keep_geom_type(multi_poly_gdf, single_rectangle_gdf):
     """Test a multi poly object where the return includes a sliver.
     Also the bounds of the object should == the bounds of the clip object
     if they fully overlap (as they do in these fixtures)."""
-    clip = gpd.clip(multi_poly_gdf, single_rectangle_gdf, drop_slivers=True)
+    clip = gpd.clip(multi_poly_gdf, single_rectangle_gdf, keep_geom_type=True)
     assert hasattr(clip, "geometry")
     assert np.array_equal(clip.total_bounds, single_rectangle_gdf.total_bounds)
     # Assert returned data is a not geometry collection
     assert (clip.geom_type == "Polygon").any()
 
 
 def test_clip_multipoly_warning(multi_poly_gdf, single_rectangle_gdf):
-    """Test that a user warning is provided when clip outputs a sliver."""
+    """Test that a user warning is provided when clip outputs an extra geometry."""
     with pytest.warns(UserWarning):
         gpd.clip(multi_poly_gdf, single_rectangle_gdf)
 
 
-def test_clip_single_multipoly_no_slivers(
+def test_clip_single_multipoly_no_extra_geoms(
     buffered_locations, larger_single_rectangle_gdf
 ):
-    """Test clipping a multi poly with another poly no sliver
-    shapes are returned in this clip operation."""
+    """Test clipping a multi poly with another poly no extra
+    geometry types are returned in this clip operation."""
     multi = buffered_locations.dissolve(by="type").reset_index()
```

Comment:
When clipping a multi-polygon feature, no additional geom types should be returned